spring-beans/src/main/java/org/springframework/beans/factory/support/ConstructorResolver.java

```diff
@@ -809,7 +809,7 @@ private ArgumentsHolder createArgumentArray(
 		for (int argIndex = 0; argIndex < argsToResolve.length; argIndex++) {
 			Object argValue = argsToResolve[argIndex];
 			MethodParameter methodParam = MethodParameter.forExecutable(executable, argIndex);
-			if (argValue instanceof AutowiredArgumentMarker) {
+			if (argValue == AutowiredArgumentMarker.INST) {
```

Comment:
1. This may save several CPU steps. But it makes the code less understandable. Many developers will not understand, why some values are compared by value, the other by type.

2. Besides, in the future this may be extended to handle special cases like autowired on setter or autowired on constructor or autowired with parameters like `@Autowired(required = true)`. That's why I would prefer keeping  the type, not the value.

Reply:
Thanks for the review, @mentallurg.

Regarding point 1, I don't think many developers will actually be reading those lines of code, and those that do can now read the improved Javadoc in 4cd635e9c87d58b990a7b19125fdfe53007ee648.

Regarding point 2, this marker technique has not been changed in over 11 years (potentially longer). So I don't foresee it changing in the near future, but even if it does change, we can revisit this section of code at that time.